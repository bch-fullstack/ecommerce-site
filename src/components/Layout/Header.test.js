import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import { BrowserRouter } from 'react-router-dom'

it('', () => {
    // const wrapper = shallow(<Header/>);
    // console.log(wrapper.debug())

    const tree = renderer
        .create(<BrowserRouter><Header/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})