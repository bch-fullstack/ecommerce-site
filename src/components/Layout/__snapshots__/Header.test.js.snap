// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<header
  className="header mb-5"
>
  <div
    id="top-element"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-6 offer mb-3 mb-lg-0"
        >
          <a
            className="btn btn-success btn-sm"
            href="/"
            onClick={[Function]}
          >
            Offer of the day
          </a>
          <a
            className="ml-1"
            href="/"
            onClick={[Function]}
          >
            Get flat 35% off on orders over $50!
          </a>
        </div>
        <div
          className="col-lg-6 text-center text-lg-right"
        >
          <ul
            className="menu list-inline mb-0"
          >
            <li
              className="list-inline-item"
            >
              <a
                data-target="#login-modal"
                data-toggle="modal"
                href="/"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
            <li
              className="list-inline-item"
            >
              <a
                href="/register.html"
                onClick={[Function]}
              >
                Register
              </a>
            </li>
            <li
              className="list-inline-item"
            >
              <a
                href="/contact.html"
                onClick={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="list-inline-item"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Recently viewed
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="Login"
      className="modal fade"
      id="login-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        className="modal-dialog modal-sm"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
            >
              Customer login
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <form
              action="customer-orders.html"
              method="post"
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="email-modal"
                  placeholder="email"
                  type="text"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="password-modal"
                  placeholder="password"
                  type="password"
                />
              </div>
              <p
                className="text-center"
              >
                <button
                  className="btn btn-primary"
                >
                  <i
                    className="fa fa-sign-in"
                  />
                   Log in
                </button>
              </p>
            </form>
            <p
              className="text-center text-muted"
            >
              Not registered yet?
            </p>
            <p
              className="text-center text-muted"
            >
              <a
                href="/register.html"
                onClick={[Function]}
              >
                <strong>
                  Register now
                </strong>
              </a>
              ! It is easy and done in 1 minute and gives you access to special discounts and much more!
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <nav
    className="navbar navbar-expand-lg"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand home"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Obaju logo"
          className="d-none d-md-inline-block"
          src="img/logo.png"
        />
        <img
          alt="Obaju logo"
          className="d-inline-block d-md-none"
          src="img/logo-small.png"
        />
        <span
          className="sr-only"
        >
          Obaju - go to homepage
        </span>
      </a>
      <div
        className="navbar-buttons"
      >
        <button
          className="btn btn-outline-secondary navbar-toggler"
          data-target="#navigation"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <i
            className="fa fa-align-justify"
          />
        </button>
        <button
          className="btn btn-outline-secondary navbar-toggler"
          data-target="#search"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle search
          </span>
          <i
            className="fa fa-search"
          />
        </button>
        <a
          className="btn btn-outline-secondary navbar-toggler"
          href="/basket.html"
          onClick={[Function]}
        >
          <i
            className="fa fa-shopping-cart"
          />
        </a>
      </div>
      <div
        className="collapse navbar-collapse"
        id="navigation"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item dropdown menu-large"
          >
            <a
              className="dropdown-toggle nav-link"
              data-delay="200"
              data-hover="dropdown"
              data-toggle="dropdown"
              href="/"
              onClick={[Function]}
            >
              Men
              <b
                className="caret"
              />
            </a>
            <ul
              className="dropdown-menu megamenu"
            >
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Clothing
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          T-shirts
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Shirts
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Pants
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Accessories
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Shoes
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Accessories
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Featured
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                    </ul>
                    <h5>
                      Looks and trends
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </li>
            </ul>
          </li>
          <li
            className="nav-item dropdown menu-large"
          >
            <a
              className="dropdown-toggle nav-link"
              data-delay="200"
              data-hover="dropdown"
              data-toggle="dropdown"
              href="/"
              onClick={[Function]}
            >
              Ladies
              <b
                className="caret"
              />
            </a>
            <ul
              className="dropdown-menu megamenu"
            >
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Clothing
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          T-shirts
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Shirts
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Pants
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Accessories
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Shoes
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Accessories
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Casual
                        </a>
                      </li>
                    </ul>
                    <h5>
                      Looks and trends
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Trainers
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Sandals
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Hiking shoes
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <div
                      className="banner"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt=""
                          className="img img-fluid"
                          src="img/banner.jpg"
                        />
                      </a>
                    </div>
                    <div
                      className="banner"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt=""
                          className="img img-fluid"
                          src="img/banner2.jpg"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </li>
          <li
            className="nav-item dropdown menu-large"
          >
            <a
              className="dropdown-toggle nav-link"
              data-delay="200"
              data-hover="dropdown"
              data-toggle="dropdown"
              href="/"
              onClick={[Function]}
            >
              Template
              <b
                className="caret"
              />
            </a>
            <ul
              className="dropdown-menu megamenu"
            >
              <li>
                <div
                  className="row"
                >
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Shop
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/"
                          onClick={[Function]}
                        >
                          Homepage
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category.html"
                          onClick={[Function]}
                        >
                          Category - sidebar left
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category-right.html"
                          onClick={[Function]}
                        >
                          Category - sidebar right
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/category-full.html"
                          onClick={[Function]}
                        >
                          Category - full width
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/detail.html"
                          onClick={[Function]}
                        >
                          Product detail
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      User
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/register.html"
                          onClick={[Function]}
                        >
                          Register / login
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/customer-orders.html"
                          onClick={[Function]}
                        >
                          Orders history
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/customer-order.html"
                          onClick={[Function]}
                        >
                          Order history detail
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/customer-wishlist.html"
                          onClick={[Function]}
                        >
                          Wishlist
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/customer-account.html"
                          onClick={[Function]}
                        >
                          Customer account / change password
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Order process
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/basket.html"
                          onClick={[Function]}
                        >
                          Shopping cart
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/checkout1.html"
                          onClick={[Function]}
                        >
                          Checkout - step 1
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/checkout2.html"
                          onClick={[Function]}
                        >
                          Checkout - step 2
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/checkout3.html"
                          onClick={[Function]}
                        >
                          Checkout - step 3
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/checkout4.html"
                          onClick={[Function]}
                        >
                          Checkout - step 4
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="col-md-6 col-lg-3"
                  >
                    <h5>
                      Pages and blog
                    </h5>
                    <ul
                      className="list-unstyled mb-3"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/blog.html"
                          onClick={[Function]}
                        >
                          Blog listing
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/post.html"
                          onClick={[Function]}
                        >
                          Blog Post
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/faq.html"
                          onClick={[Function]}
                        >
                          FAQ
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/text.html"
                          onClick={[Function]}
                        >
                          Text page
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/text-right.html"
                          onClick={[Function]}
                        >
                          Text page - right sidebar
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/404.html"
                          onClick={[Function]}
                        >
                          404 page
                        </a>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link"
                          href="/contact.html"
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </li>
            </ul>
          </li>
        </ul>
        <div
          className="navbar-buttons d-flex justify-content-end"
        >
          <div
            className="navbar-collapse collapse"
            id="search-not-mobile"
          />
          <a
            className="btn navbar-btn btn-primary d-none d-lg-inline-block"
            data-toggle="collapse"
            href="/#search"
            onClick={[Function]}
          >
            <span
              className="sr-only"
            >
              Toggle search
            </span>
            <i
              className="fa fa-search"
            />
          </a>
          <div
            className="navbar-collapse collapse d-none d-lg-block"
            id="basket-overview"
          >
            <a
              className="btn btn-primary navbar-btn"
              href="/basket.html"
              onClick={[Function]}
            >
              <i
                className="fa fa-shopping-cart"
              />
              <span>
                3 items in cart
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="collapse"
    id="search"
  >
    <div
      className="container"
    >
      <form
        className="ml-auto"
        role="search"
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            placeholder="Search"
            type="text"
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-primary"
              type="button"
            >
              <i
                className="fa fa-search"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</header>
`;
